core/src/main/scala/org/apache/spark/deploy/master/ui/MasterWebUI.scala

```diff
@@ -34,7 +34,12 @@ class MasterWebUI(
   extends WebUI(master.securityMgr, master.securityMgr.getSSLOptions("standalone"),
     requestedPort, master.conf, name = "MasterUI") with Logging {
 
-  val masterEndpointRef = master.self
+  val masterEndpointRef = {
```

Comment:
If we don't feel comfortable adding infinite loop, we can just change it from val to def, with adding comment it shouldn't be cached.